File: app/src/main/java/com/kota/Telnet/TelnetClient.java
package com.kota.Telnet;

import android.util.Log;

import com.kota.ASFramework.Thread.ASRunner;
import com.kota.Bahamut.DataModels.UrlDatabase;
import com.kota.Bahamut.MyApplication;
import com.kota.Bahamut.Service.TempSettings;
import com.kota.Telnet.Model.TelnetModel;
import com.kota.Telnet.Reference.TelnetDefs;
import com.kota.Telnet.Reference.TelnetKeyboard;
import com.kota.TextEncoder.U2BEncoder;
import java.io.UnsupportedEncodingException;
import java.util.Objects;
import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;

/* loaded from: classes.dex */
public class TelnetClient implements TelnetConnectorListener {
    private static TelnetClient _instance = null;
    private TelnetConnector _connector;
    private TelnetModel _model;
    private TelnetReceiver _receiver;
    private final TelnetStateHandler _state_handler;
    private String _username;
    private TelnetClientListener _listener = null;
    ExecutorService _send_executor = Executors.newSingleThreadExecutor();

    public static void construct(TelnetStateHandler aStateHandler) {
        _instance = new TelnetClient(aStateHandler);
    }

    public static TelnetClient getClient() {
        return _instance;
    }

    public static TelnetConnector getConnector() {
        return getClient()._connector;
    }

    public static TelnetModel getModel() {
        return getClient()._model;
    }

    private TelnetClient(TelnetStateHandler aStateHandler) {
        _connector = null;
        _receiver = null;
        _model = null;
        _state_handler = aStateHandler;
        _model = new TelnetModel();
        _connector = new TelnetConnector();
        _connector.setListener(this);
        _receiver = new TelnetReceiver(_connector, _model);
    }

    public void clear() {
        _state_handler.clear();
        _connector.clear();
        _model.clear();
        _receiver.stopReceiver();
    }

    public void connect(String serverIp, int serverPort) {
        _connector.connect(serverIp, serverPort);
    }

    public void close() {
        try {
            _connector.close(); // 關閉連線
        } catch (Exception e) {
            Log.e(getClass().getSimpleName(), Objects.requireNonNull(e.getMessage()));
        }
    }

    public void sendStringToServer(String str) {
        if (ASRunner.isMainThread()) {
            sendStringToServerInBackground(str, 0);
        } else {
            sendStringToServer(str, 0);
        }
    }

    private void sendStringToServer(String str, int channel) {
        byte[] data = null;
        boolean encode_success = false;
        try {
            byte[] data2 = (str + "\n").getBytes(TelnetDefs.CHARSET);
            data = U2BEncoder.getInstance().encodeToBytes(data2, 0);
            encode_success = true;
        } catch (UnsupportedEncodingException e) {
            Log.e(getClass().getSimpleName(), Objects.requireNonNull(e.getMessage()));
        }
        if (encode_success) {
            sendDataToServer(data, channel);
        }
    }

    public void sendStringToServerInBackground(String str) {
        sendStringToServerInBackground(str, 0);
    }

    public void sendStringToServerInBackground(String str, int channel) {
        byte[] data = null;
        boolean encode_success = false;
        try {
            byte[] data2 = (str + "\n").getBytes(TelnetDefs.CHARSET);
            data = U2BEncoder.getInstance().encodeToBytes(data2, 0);
            encode_success = true;
        } catch (UnsupportedEncodingException e) {
            Log.e(getClass().getSimpleName(), Objects.requireNonNull(e.getMessage()));
        }
        if (encode_success) {
            sendDataToServerInBackground(data, channel);
        }
    }

    public void sendKeyboardInputToServer(int key) {
        if (ASRunner.isMainThread()) {
            sendKeyboardInputToServerInBackground(key, 0);
        } else {
            sendKeyboardInputToServer(key, 0);
        }
    }

    private void sendKeyboardInputToServer(int key, int channel) {
        sendDataToServer(TelnetKeyboard.getKeyData(key), channel);
    }

    public void sendKeyboardInputToServerInBackground(int key) {
        sendKeyboardInputToServerInBackground(key, 0);
    }

    public void sendKeyboardInputToServerInBackground(int key, int channel) {
        sendDataToServerInBackground(TelnetKeyboard.getKeyData(key), channel);
    }

    public void sendDataToServer(byte[] data) {
        if (ASRunner.isMainThread()) {
            sendDataToServerInBackground(data, 0);
        } else {
            sendDataToServer(data, 0);
        }
    }

    protected void sendDataToServer(byte[] data, int channel) {
        if (data != null && _connector.isConnecting()) {
            _connector.writeData(data, channel);
            _connector.sendData(channel);
        }
    }

    public void sendDataToServerInBackground(final byte[] data, final int channel) {
        if (data != null && _connector.isConnecting()) {
            _send_executor.submit(() -> {
                TelnetClient.this._connector.writeData(data, channel);
                TelnetClient.this._connector.sendData(channel);
            });
        }
    }

    public void setListener(TelnetClientListener aListener) {
        _listener = aListener;
    }

    @Override // com.kota.Telnet.TelnetConnectorListener
    public void onTelnetConnectorConnectStart(TelnetConnector aConnector) {
        if (_listener != null) {
            _listener.onTelnetClientConnectionStart(this);
        }
    }

    @Override // com.kota.Telnet.TelnetConnectorListener
    public void onTelnetConnectorClosed(TelnetConnector aConnector) {
        clear();
        if (_listener != null) {
            _listener.onTelnetClientConnectionClosed(this);
        }
    }

    @Override // com.kota.Telnet.TelnetConnectorListener
    public void onTelnetConnectorConnectSuccess(TelnetConnector aConnector) {
        _receiver.startReceiver();
        if (_listener != null) {
            _listener.onTelnetClientConnectionSuccess(this);
        }
    }

    @Override // com.kota.Telnet.TelnetConnectorListener
    public void onTelnetConnectorConnectFail(TelnetConnector aConnector) {
        clear();
        if (_listener != null) {
            _listener.onTelnetClientConnectionFail(this);
        }
    }

    @Override // com.kota.Telnet.TelnetConnectorListener
    public void onTelnetConnectorReceiveDataStart(TelnetConnector aConnector) {
        if (_state_handler != null) {
            _model.cleanCahcedData();
            _state_handler.handleState();
        }
    }

    @Override // com.kota.Telnet.TelnetConnectorListener
    public void onTelnetConnectorReceiveDataFinished(TelnetConnector aConnector) {
        _connector.cleanReadDataSize();
    }

    public void setUsername(String aUsername) {
        _username = aUsername;
    }

    public String getUsername() {
        if (_username == null) {
            _username = "";
        }
        return _username;
    }
}

File: app/src/main/java/com/kota/telnet/TelnetClient.kt
package com.kota.telnet

import android.util.Log
import com.kota.asFramework.thread.ASRunner
import com.kota.telnet.model.TelnetModel
import com.kota.telnet.reference.TelnetDef
import com.kota.telnet.reference.TelnetKeyboard
import com.kota.textEncoder.U2BEncoder
import java.io.UnsupportedEncodingException
import java.util.concurrent.ExecutorService
import java.util.concurrent.Executors

class TelnetClient(aStateHandler: TelnetStateHandler) : TelnetConnectorListener {
    var telnetConnector: TelnetConnector? = null
    private var telnetModel: TelnetModel? = null
    private var telnetReceiver: TelnetReceiver? = null
    private var stateHandler: TelnetStateHandler? = null
    private var myUsername: String = ""
    private var telnetClientListener: TelnetClientListener? = null
    var executorService: ExecutorService = Executors.newSingleThreadExecutor()

    // 初始化塊 - 在主構造器之後執行
    init {
        stateHandler = aStateHandler
        telnetModel = TelnetModel()
        telnetConnector = TelnetConnector()
        telnetConnector?.setListener(this)
        telnetReceiver = TelnetReceiver(telnetConnector, telnetModel)
    }

    fun clear() {
        stateHandler!!.clear()
        telnetConnector!!.clear()
        telnetModel!!.clear()
        telnetReceiver!!.stopReceiver()
    }

    fun connect(serverIp: String?, serverPort: Int) {
        telnetConnector!!.connect(serverIp, serverPort)
    }

    fun close() {
        try {
            telnetConnector!!.close() // 關閉連線
        } catch (e: Exception) {
            Log.e(javaClass.simpleName, (if (e.message != null) e.message else "")!!)
        }
    }

    fun sendStringToServer(str: String?) {
        if (ASRunner.isMainThread) {
            sendStringToServerInBackground(str, 0)
        } else {
            sendStringToServer(str, 0)
        }
    }

    private fun sendStringToServer(str: String?, channel: Int) {
        var data: ByteArray? = null
        var encodeSuccess = false
        try {
            val data2: ByteArray = (str + "\n").toByteArray(charset(TelnetDef.CHARSET))
            data = U2BEncoder.instance!!.encodeToBytes(data2, 0)
            encodeSuccess = true
        } catch (e: UnsupportedEncodingException) {
            Log.e(javaClass.simpleName, (if (e.message != null) e.message else "")!!)
        }
        if (encodeSuccess) {
            sendDataToServer(data, channel)
        }
    }

    @JvmOverloads
    fun sendStringToServerInBackground(str: String?, channel: Int = 0) {
        var data: ByteArray? = null
        var encodeSuccess = false
        try {
            val data2: ByteArray = (str + "\n").toByteArray(charset(TelnetDef.CHARSET))
            data = U2BEncoder.instance!!.encodeToBytes(data2, 0)
            encodeSuccess = true
        } catch (e: UnsupportedEncodingException) {
            Log.e(javaClass.simpleName, (if (e.message != null) e.message else "")!!)
        }
        if (encodeSuccess) {
            sendDataToServerInBackground(data, channel)
        }
    }

    fun sendKeyboardInputToServer(key: Int) {
        if (ASRunner.isMainThread) {
            sendKeyboardInputToServerInBackground(key, 0)
        } else {
            sendKeyboardInputToServer(key, 0)
        }
    }

    private fun sendKeyboardInputToServer(key: Int, channel: Int) {
        sendDataToServer(TelnetKeyboard.getKeyData(key), channel)
    }

    @JvmOverloads
    fun sendKeyboardInputToServerInBackground(key: Int, channel: Int = 0) {
        sendDataToServerInBackground(TelnetKeyboard.getKeyData(key), channel)
    }

    fun sendDataToServer(data: ByteArray?) {
        if (ASRunner.isMainThread) {
            sendDataToServerInBackground(data, 0)
        } else {
            sendDataToServer(data, 0)
        }
    }

    fun sendDataToServer(data: ByteArray?, channel: Int) {
        if (data != null && telnetConnector!!.isConnecting) {
            telnetConnector!!.writeData(data, channel)
            telnetConnector!!.sendData(channel)
        }
    }

    fun sendDataToServerInBackground(data: ByteArray?, channel: Int) {
        if (data != null && telnetConnector!!.isConnecting) {
            executorService.submit {
                this@TelnetClient.telnetConnector!!.writeData(data, channel)
                this@TelnetClient.telnetConnector!!.sendData(channel)
            }
        }
    }

    fun setListener(aListener: TelnetClientListener?) {
        telnetClientListener = aListener
    }

    // com.kota.Telnet.TelnetConnectorListener
    override fun onTelnetConnectorConnectStart(telnetConnector: TelnetConnector) {
        telnetClientListener?.onTelnetClientConnectionStart(this)
    }

    // com.kota.Telnet.TelnetConnectorListener
    override fun onTelnetConnectorClosed(telnetConnector: TelnetConnector) {
        clear()
        telnetClientListener?.onTelnetClientConnectionClosed(this)
    }

    // com.kota.Telnet.TelnetConnectorListener
    override fun onTelnetConnectorConnectSuccess(telnetConnector: TelnetConnector) {
        telnetReceiver!!.startReceiver()
        telnetClientListener?.onTelnetClientConnectionSuccess(this)
    }

    // com.kota.Telnet.TelnetConnectorListener
    override fun onTelnetConnectorConnectFail(telnetConnector: TelnetConnector) {
        clear()
        telnetClientListener?.onTelnetClientConnectionFail(this)
    }

    // com.kota.Telnet.TelnetConnectorListener
    override fun onTelnetConnectorReceiveDataStart(telnetConnector: TelnetConnector) {
        if (stateHandler != null) {
            telnetModel!!.cleanCachedData()
            stateHandler!!.handleState()
        }
    }

    // com.kota.Telnet.TelnetConnectorListener
    override fun onTelnetConnectorReceiveDataFinished(telnetConnector: TelnetConnector) {
        this@TelnetClient.telnetConnector!!.cleanReadDataSize()
    }

    var username: String
        get() {
            return myUsername
        }
        set(aUsername) {
            myUsername = aUsername
        }

    companion object {
        var myInstance: TelnetClient? = null
        fun construct(aStateHandler: TelnetStateHandler) {
            myInstance = TelnetClient(aStateHandler)
        }

        val client: TelnetClient?
            get() = myInstance

        val connector: TelnetConnector
            get() = client!!.telnetConnector!!

        val model: TelnetModel
            get() = client!!.telnetModel!!
    }
}
